compiler+runtime/src/cpp/clojure/core_native.cpp

```diff
@@ -578,7 +578,8 @@ jank_object_ptr jank_load_clojure_core_native()
         return obj::boolean::false_const();
       }
 
-      for(auto it(fresh_seq(rest)); it != nullptr; it = next_in_place(it))
+      for(auto it(fresh_seq(rest)); it != nullptr && it != obj::nil::nil_const();
```

Comment:
So, here, `it` cannot be `nullptr`. That's not part of the contract. Everything returned from `runtime::` fns should be non-null.

The one exception is the `next_in_place` and `fresh_seq` templates. Ultimately, we can remove those (not in this PR); they're only there for a historical partial refactor which never panned out.

There should never be a case, right now, where we need to check for nullptr *and* nil.

Reply:
> There should never be a case, right now, where we need to check for nullptr and nil.

There's a few more in `multi_function.cpp`.